js/apps/system/_admin/aardvark/APP/frontend/js/collections/arangoDocument.js

```diff
@@ -69,10 +67,10 @@ window.ArangoDocument = Backbone.Collection.extend({
     });
   },
 
-  createTypeDocument: function (collectionID, key, callback, returnNew,
+  createTypeDocument: function (collectionID, key, body, callback, returnNew,
                                 smartJoinAttribute, smartJoinAttributeValue,
                                 smartGraphAttribute, smartGraphAttributeValue) {
-    var newDocument = {};
+    var newDocument = body;
```

Comment:
Same as above. Just maybe add a method e.g. "verifyBody".
Fine, but maybe add a check that the body is a valid object. Otherwise 
```
body._key = key;
```
and so on will fail.